sql/core/src/main/java/org/apache/spark/sql/execution/BufferedRowIterator.java

```diff
@@ -65,7 +65,7 @@ public long durationMs() {
   /**
    * Append a row to currentRows.
    */
-  protected void append(InternalRow row) {
+  public void append(InternalRow row) {
```

Comment:
nit: Although we added the test, should we also add a short sentence that said this is public so inner classes can also access it?